pincer/objects/guild/guild.py

```diff
@@ -1916,6 +1919,133 @@ async def get_webhooks(self) -> AsyncGenerator[Webhook, None]:
+            f"guilds/{self.id}/scheduled-events",
+            data={
+                "name": name,
+                "privacy_level": privacy_level,
+                "entity_type": entity_type,
+                "scheduled_start_time": scheduled_start_time.isoformat(),
+                "scheduled_end_time": scheduled_end_time.isoformat()
+                if scheduled_end_time is not None
+                else None,
+                "entity_metadata": entity_metadata,
+                "channel_id": channel_id,
+                "description": description,
+            },
+            headers={"X-Audit-Log-Reason": reason},
+        )
+        return ScheduledEvent.from_dict(data)
+
+    async def get_scheduled_event(
+        self, _id: Snowflake, with_user_count: bool = False
+    ) -> ScheduledEvent:
+        data = await self._http.get(
+            f"guilds/{self.id}/scheduled-events/{_id}?{with_user_count=!s}"
+        )
+        return ScheduledEvent.from_dict(self._client, data)
+
+
+    async def modify_scheduled_event(
+        self,
+        _id: Snowflake,
+        name: Optional[str] = None,
+        entity_type: Optional[int] = None,
+        privacy_level: Optional[int] = None,
+        scheduled_start_time: Optional[datetime] = None,
+        scheduled_end_time: Optional[datetime] = None,
+        entity_metadata: Optional[str] = None,
+        channel_id: Optional[Snowflake] = None,
+        description: Optional[str] = None,
+        status: Optional[int] = None,
+        reason: Optional[str] = None,
+    ) -> ScheduledEvent:
+        if scheduled_start_time:
+            timestamp = scheduled_start_time.timestamp()
+            if timestamp < datetime.now().timestamp():
+                raise ValueError("An event cannot be created in the past")
+
+            if scheduled_end_time and timestamp > scheduled_end_time.timestamp():
+                raise ValueError("An event cannot start before it ends")
+
+        kwargs: Dict[str, str] = remove_none(
+            {
+                "name": name,
+                "privacy_level": privacy_level,
+                "entity_type": entity_type,
+                "scheduled_start_time": scheduled_start_time.isoformat()
+                if scheduled_start_time is not None
+                else None,
+                "scheduled_end_time": scheduled_end_time.isoformat()
+                if scheduled_end_time is not None
+                else None,
+                "entity_metadata": entity_metadata,
+                "channel_id": channel_id,
+                "description": description,
+                "status": status,
+            }
+        )
+
+        data = await self._http.patch(
+            f"guilds/{self.id}/scheduled-events/{_id}",
+            data=kwargs,
+            headers={"X-Audit-Log-Reason": reason},
+        )
+        return ScheduledEvent.from_dict(self._client, data)
+
+    async def delete_scheduled_event(self, _id: Union[int, Snowflake]):
```

Comment:
I don't think you need to specify int if snowflake is there. No sure though.

Reply:
A Snowflake is an int but an int is not necessarily a Snowflake, so yes?

Reply:
# Typing with Snowflake
```py
def bar(x: Snowflake):
    print(x + 69)


bar(69)
bar(Snowflake(69))
```
> ```
> def.py:5: error: Argument 1 to "bar" has incompatible type "int"; expected "Snowflake"  [arg-type]
> Found 1 error in 1 file (checked 1 source file)
> ```


# Typing with Int
```py
def baz(x: int):
    print(x + 42)


baz(42)
baz(Snowflake(42))
```
> ```
Success: no issues found in 1 source file
>```

Reply:
Ig always type with int